torch/distributed/_tensor/_dispatch.py

```diff
@@ -107,6 +109,26 @@ def dispatch(
         """
         Main dispatching logic
         """
+        if logger.isEnabledFor(logging.DEBUG):
+            tensor_args = ""
+            for idx, arg in enumerate(args):
+                if not isinstance(arg, torch.Tensor):
+                    continue
+                if hasattr(arg, "_spec"):
+                    tensor_args += f"\n  arg_{idx} (dtensor): {arg.shape=} {arg._spec=}"
+                else:
+                    tensor_args += f"\n  arg_{idx} (tensor): {arg.shape=}"
+
+            for key, arg in kwargs.items():
+                if not isinstance(arg, torch.Tensor):
+                    continue
+                if hasattr(arg, "_spec"):
+                    tensor_args += f"\n  arg_{key} (dtensor): {arg.shape=} {arg._spec=}"
+                else:
+                    tensor_args += f"\n  arg_{key} (tensor): {arg.shape=}"
```

Comment:
I wonder why we need we need to unwrap those DTensor args here? IMO we can just print the information after unwrap_op_info call?

Reply:
Good call, I moved the block and use the `op_info`.